plugin-server/src/worker/ingestion/team-manager.ts

```diff
@@ -143,79 +164,32 @@ export class TeamManager {
             }
         }
     }
-}
 
-export async function fetchTeam(client: PostgresRouter, teamId: Team['id']): Promise<Team | null> {
-    const selectResult = await client.query<Team>(
-        PostgresUse.COMMON_READ,
-        `
-            SELECT
-                id,
-                project_id,
-                uuid,
-                organization_id,
-                name,
-                anonymize_ips,
-                api_token,
-                slack_incoming_webhook,
-                session_recording_opt_in,
-                person_processing_opt_out,
-                heatmaps_opt_in,
-                ingested_event,
-                person_display_name_properties,
-                test_account_filters,
-                cookieless_server_hash_mode,
-                timezone
-            FROM posthog_team
-            WHERE id = $1
+    private async fetchTeamByTokenOrId(params: { token?: string; id?: number }): Promise<Team | null> {
+        if (!params.token && !params.id) {
+            return null
+        }
+
+        const result = await this.postgres.query<Team>(
```

Comment:
thought: The team schema is fairly complex (test_account_filters), I'm always a bit worried about getting unexpected values back from the DB (e.g. by someone changing the writers and forgetting to update the readers) and spreading damage throughout the rest of the code. WDYT about using zod to verify the rows returned by the query? We could model the returned object fairly accurately, or at least make it clear which parts we haven't validated. It's a little defensive approach, but it comes at a pretty low cost.

Reply:
Not _against_ it but also not super for it. Would rather have tests cover our back for hyper deterministic code like this than add a bunch of cpu cycles for nothing

Reply:
In any case definitely not making those changes here - trying to do small PRs :D

Reply:
ah crap I pushed other changes to this branch... will fix that...